src/Symfony/Component/AstGenerator/AstGeneratorInterface.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\AstGenerator;
+
+/**
+ * An AstGeneratorInterface is a contract to transform an object into an AST.
+ *
+ * @author Joel Wurtz <joel.wurtz@gmail.com>
+ */
+interface AstGeneratorInterface
+{
+    /**
+     * Generate an object into an AST given a specific context.
+     *
+     * @param mixed $object  Object to generate AST from
+     * @param array $context Context for the generator
+     *
+     * @return \PhpParser\Node[] An array of statements (AST Node)
+     */
+    public function generate($object, array $context = array());
+
+    /**
+     * Check whether the given object is supported for generation by this generator.
+     *
+     * @param mixed $object Object to generate AST from
+     *
+     * @return bool
+     */
+    public function supportsGeneration($object);
```

Comment:
>This method is especially useful for chaining

Can't the chain be done with decoration only by throwing a `UnsupportedObjectException`? I'm not sure if having a chain mechanism based on exceptions would be slower or not, but it would avoid to have two public methods and use `generate()` without trying `supportsGeneration()`.